### PR TITLE
Reduced Quarkus image size

### LINKED ISSUE #10073: Reduce Keycloak.x image size
### Description

The current multi-layer build effectively adds the Keycloak Quarkus distribution a second time due to the `chown` and `chmod`. This could be improved by moving this step to the previous layer so the Keycloak distribution will effectively only created once. See screenshot of the image comparison.
Screenshot from 2022-02-08 16-34-41
.

### Discussion

_No response_

### Motivation

Faster container pull/initialization times, less space usage.

### Details

_No response_

### PR DESCRIPTION
**Pull request by sschu**
Closes #10073

### CODE REVIEW

**Review by andreaTP — DISMISSED** [review]
LGTM

**Comment by sschu** [comment]
The test run failure might be related to my change, I will have a look...

**Comment by sschu** [comment]
Alright, looks like COPY always uses root as owner unless you tell it to do otherwise. Permissions are taken over. Fixed that.

**Comment by andreaTP** [comment]
Happy to see that the Quarkus integration tests on Docker are effective! 🎉

**Comment by sschu** [comment]
I highly doubt the last build failure has to do with my changes. Is there any way I can trigger another build?

**Review by DGuhr — DISMISSED** [review]

**Review by stianst — CHANGES_REQUESTED** [review]
Just tested this out here with podman and getting the following error:
`
Error: error building at STEP "COPY --from=build-env --chown 1000:0 /opt/keycloak /opt/keycloak": COPY only supports the --chmod=<permissions> --chown=<uid:gid> and the --from=<image|stage> flags
`

File: quarkus/container/Dockerfile [hunk 1]
```diff
@@ -15,16 +15,16 @@ RUN (cd /tmp/keycloak && \
 
 RUN mv /tmp/keycloak/keycloak-* /opt/keycloak
 
+RUN chmod -R g+rwX /opt/keycloak
+
 FROM registry.access.redhat.com/ubi8-minimal
 
-COPY --from=build-env /opt/keycloak /opt/keycloak
+COPY --from=build-env --chown 1000:0 /opt/keycloak /opt/keycloak
```
Comment (stianst) [inline quarkus/container/Dockerfile:22]:
```suggestion
COPY --from=build-env --chown=1000:0 /opt/keycloak /opt/keycloak
```

**Review by stianst — APPROVED** [review]

**Review by DGuhr — APPROVED** [review]

**Review by pedroigor — APPROVED** [review]

**Comment by sschu** [comment]
Strange, its always the same error in the build:
`Could not transfer artifact org.picketbox:picketbox:pom:5.0.3.Final-redhat-00008 from/to jboss-enterprise-maven-repository (https://maven.repository.redhat.com/ga/): transfer failed for https://maven.repository.redhat.com/ga/org/picketbox/picketbox/5.0.3.Final-redhat-00008/picketbox-5.0.3.Final-redhat-00008.pom, status: 500 Internal Server Error` looks like the repository has a problem.

**Pull request merged**